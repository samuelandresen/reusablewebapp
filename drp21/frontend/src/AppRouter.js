import React, { useMemo, useState } from 'react';
import Feed from './pages/Forum/Feed';
import LoginPage from './pages/Login/LoginPage';
import Marketplace from './pages/MarketPlace/Marketplace';
import NavBar from './pages/NavBar';
import Home from './pages/Home/Home'
import CreateAccount from './pages/Login/CreateAccount';
import ReqHome from './pages/Request/ReqHome'
import { BrowserRouter, Route } from 'react-router-dom';
import { UserContext } from './UserContext';
import { RepairPost } from './RepairPost';
import { RequestPost } from './RequestPost';
import { ListingPost } from './ListingPost';


function AppRouter() {
  const [user, setUser] = useState(null);
  const [post, setPosts] = useState(null)
  const [request, setRequests] = useState(null)
  const [listing, setListings] = useState(null)

  const userMemo = useMemo(() => ({ user, setUser }), [user, setUser])
  const postMemo = useMemo(() => ({ post, setPosts }), [post, setPosts])
  const requestMemo = useMemo(() => ({ request, setRequests }), [request, setRequests])
  const listingMemo = useMemo(() => ({ listing, setListings }), [listing, setListings])


  return (
    <BrowserRouter>
      <ListingPost.Provider value={listingMemo}>
        <RepairPost.Provider value={postMemo}>
          <RequestPost.Provider value={requestMemo}>
            <UserContext.Provider value={userMemo}>
              <NavBar />
              <Route exact path='/' component={Home} />
              <Route exact path='/requests' component={ReqHome} />
              <Route exact path='/feed' component={Feed} />
              <Route exact path='/market' component={Marketplace} />
              <Route exact path='/login' component={LoginPage} />
              <Route exact path='/createAccount' component={CreateAccount} />
            </UserContext.Provider>
          </RequestPost.Provider>
        </RepairPost.Provider>
      </ListingPost.Provider>
    </BrowserRouter>
  );
}

export default AppRouter;