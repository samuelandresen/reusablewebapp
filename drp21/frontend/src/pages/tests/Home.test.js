import renderer from 'react-test-renderer';
import AppRouter from '../../AppRouter';


describe('NavBar component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<AppRouter/>).toJSON()

        expect(tree).toMatchSnapshot()
    })
})