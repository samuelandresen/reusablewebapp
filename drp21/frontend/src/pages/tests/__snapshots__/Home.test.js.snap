// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches the snapshot 1`] = `
Array [
  <div
    className="NavBar"
  >
    <p>
      <a
        className="NavBarLink"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="NavBarLink"
        href="/market"
        onClick={[Function]}
      >
        Marketplace
      </a>
      <a
        className="NavBarLink"
        href="/requests"
        onClick={[Function]}
      >
        Requests
      </a>
      <a
        className="NavBarLink"
        href="/feed"
        onClick={[Function]}
      >
        Repair Forum
      </a>
      <div
        className="UserSection"
      >
        <button
          className="LogoutBtn"
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </p>
  </div>,
  <div
    className="Home"
  >
    <header
      className="Home-header"
    >
      <div
        className="HomeContainer"
        style={
          Object {
            "width": "90%",
          }
        }
      >
        <img
          alt="logo"
          className="Home-logo"
          src="recycle-logo.png"
        />
        <h3
          className="AppTitle"
        >
          eCycling
        </h3>
        <div
          className="Home-redirects"
        >
          <button
            className="Home-btn"
            onClick={[Function]}
          >
            Give or get devices
          </button>
          <button
            className="Home-btn"
            onClick={[Function]}
          >
            Request some devices
          </button>
          <button
            className="Home-btn"
            onClick={[Function]}
          >
            Need help repairing
          </button>
        </div>
        <button
          className="Login-btn btn"
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </header>
  </div>,
]
`;
